src/pycatcher/diagnostics.py

```diff
@@ -379,6 +379,9 @@ def build_stl_outliers_plot(df) -> plt:
     df_stl.iloc[:, 0] = df_stl.iloc[:, 0].apply(pd.to_datetime)
     df_stl = df_stl.set_index(df_stl.columns[0]).dropna()
 
+    # Initializing df_outliers to avoid undefined usage
+    df_outliers = pd.DataFrame()
```

Comment:
Added the initialization since it was calling out: possibly-used-before-assignment / Possibly using variable 'df_outliers' before assignment (col 16)